### PR TITLE
Fix MultiAdapter

### PR DESCRIPTION
**Pull request by dotieuthien**
I face with error when I feed adapter_conditioning_scale as a list to MultiAdapter pipeline, after tracing, there are 2 problems:

- adapter_state[k] = v * adapter_conditioning_scale is just correct with single Adapter
- if accume_state is None: --> accume_state = features in MultiAdapter.forward always ignores the first weight

So, I fix the issue, and the result is quite similar with the image in your tutorial [here](https://huggingface.co/docs/diffusers/main/en/api/pipelines/stable_diffusion/adapter)

### CODE REVIEW

**Comment by dotieuthien** [comment]
Here is the result I got with the same input in tutorial
image
image

**Comment by patrickvonplaten** [comment]
@williamberman @patil-suraj can you take a look?

**Comment by sayakpaul** [comment]
Could you also see if the SDXL Adapters work for the MultiAdapter setting? They're of significantly better quality.

File: src/diffusers/models/adapter.py [hunk 1]
```diff
@@ -90,6 +90,8 @@ def forward(self, xs: torch.Tensor, adapter_weights: Optional[List[float]] = Non
             features = adapter(x)
             if accume_state is None:
                 accume_state = features
+                for i in range(len(accume_state)):
+                    accume_state[i] = w * accume_state[i]
```
Comment (patil-suraj) [inline src/diffusers/models/adapter.py:94]: This looks good to me !

File: src/diffusers/pipelines/t2i_adapter/pipeline_stable_diffusion_adapter.py [hunk 2]
```diff
@@ -727,9 +727,14 @@ def __call__(
         extra_step_kwargs = self.prepare_extra_step_kwargs(generator, eta)
 
         # 7. Denoising loop
-        adapter_state = self.adapter(adapter_input)
-        for k, v in enumerate(adapter_state):
-            adapter_state[k] = v * adapter_conditioning_scale
+        if isinstance(self.adapter, MultiAdapter):
+            adapter_state = self.adapter(adapter_input, adapter_conditioning_scale)
```
Comment (patil-suraj) [inline src/diffusers/pipelines/t2i_adapter/pipeline_stable_diffusion_adapter.py:731]: I'm a bit confused here, is `adapter_conditioning_scale` supposed to be similar to `adapter_weights` ? We are passing`adapter_weights` to `MultiAdapter` here 
https://huggingface.slack.com/archives/C05PH2CH4N5/p1693571995003459

**Review by patil-suraj — COMMENTED** [review]
Thanks a lot for the PR! The change to adapter weight looks good to me. 
But confused about `adapter_weights` and `adapter_conditioning_scale` as I mentioned in the comment above. 

Also, we pass `adapter_weights` to `MultiAdapter`, but it doesn't accept that argument. 
https://github.com/huggingface/diffusers/blob/16b9a57d29b6dbce4f97dbf439af1663d2c54588/src/diffusers/models/adapter.py#L42
cc @williamberman

File: src/diffusers/pipelines/t2i_adapter/pipeline_stable_diffusion_adapter.py (hunk 2 above)
Comment (dotieuthien) [reply src/diffusers/pipelines/t2i_adapter/pipeline_stable_diffusion_adapter.py:731]: But, now we have 2 factors that are the same manner, I m also confuse if we do that way.

File: src/diffusers/models/adapter.py (hunk 1 above)
Comment (williamberman) [inline src/diffusers/models/adapter.py:94]: ok, so the fix is that the weights are not being applied for the first adapter network, nice!

File: src/diffusers/pipelines/t2i_adapter/pipeline_stable_diffusion_adapter.py (hunk 2 above)
Comment (williamberman) [reply src/diffusers/pipelines/t2i_adapter/pipeline_stable_diffusion_adapter.py:731]: ok, yes the change in the PR is correct. The multi adapter applies the additional multiplication internally to the module and the regular adapter's results are multiplied in the pipeline. It's a bit non-uniform between the two but I think is ok

**Comment by williamberman** [comment]
> Thanks a lot for the PR! The change to adapter weight looks good to me. But confused about `adapter_weights` and `adapter_conditioning_scale` as I mentioned in the comment above.
> 
> Also, we pass `adapter_weights` to `MultiAdapter`, but it doesn't accept that argument.
> 
> https://github.com/huggingface/diffusers/blob/16b9a57d29b6dbce4f97dbf439af1663d2c54588/src/diffusers/models/adapter.py#L42
> 
> 
> cc @williamberman

I think passing in the constructor was left in on accident from the original PR. I'll just remove that.

File: src/diffusers/pipelines/t2i_adapter/pipeline_stable_diffusion_adapter.py [hunk 3]
```diff
@@ -727,9 +727,14 @@ def __call__(
...
+            for k, v in enumerate(adapter_state):
+                adapter_state[k] = v
+        else:
+            adapter_state = self.adapter(adapter_input)
+            for k, v in enumerate(adapter_state):
+                adapter_state[k] = v * adapter_conditioning_scale
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/t2i_adapter/pipeline_stable_diffusion_adapter.py:737]:
```suggestion
        if isinstance(self.adapter, MultiAdapter):
            adapter_state = self.adapter(adapter_input, adapter_conditioning_scale)
            for k, v in enumerate(adapter_state):
                adapter_state[k] = v
        else:
            adapter_state = self.adapter(adapter_input)
            for k, v in enumerate(adapter_state):
                adapter_state[k] = v * adapter_conditioning_scale
```

**Comment by patrickvonplaten** [comment]
Super-seeded by https://github.com/huggingface/diffusers/pull/5090

**Pull request closed without merging**